src/components/MessageList.tsx

```diff
@@ -0,0 +1,434 @@
+import { nanoid } from '../../node_modules/nanoid';
+
+export type Indicator =
+	 | 'indicator-unvisible'
+	 | 'indicator-edited'
+	 | 'indicator-successful'
+	 | 'indicator-unsuccessful';
+
+export type ParsedMessageItem = {
+	 sessionId: string;
+	 dictionary: string;
+	 messageType: string;
+	 message: object | string;
+	 delay: number | 0;
+ };
+
+export interface ActMessageItem {
+	 actBox: string;
+	 fullServiceName: string;
+	 methodName: string;
+	 message: object | string;
+	 delay: number | 0;
+ }
+
+ interface MessageCardControlsProps {
+	 deleteMessage: Function;
+	 editMessageMode: boolean;
+	 index: number;
+	 indicator: Indicator;
+ }
+
+ interface EditMessageProps {
+	 editMessageMode: boolean;
+	 editedMessageIndex: number;
+	 indicators: Indicator[];
+ }
+
+ interface MessageItemProps extends EditMessageProps {
+	 index: number;
+	 message: ParsedMessageItem | ActMessageItem;
+	 selectMessage: Function;
+	 setDelay: Function;
+	 deleteMessage: Function;
+ }
+
+ interface MessageListProps extends EditMessageProps {
+	 messages: ParsedMessageItem[] | ActMessageItem[];
+ }
+
+ interface MessageEditAreaProps extends EditMessageProps {
+	 messages: ParsedMessageItem[] | ActMessageItem[];
+	 eventsPanelArea: number;
+ }
+
+export const Messages = () => {
+	 const store = useStore();
+
+	 const downloadFile = (content: string, filename: string, extension: string) => {
+		 const file = new Blob([content], { type: extension });
+
+		 if (window.navigator.msSaveOrOpenBlob) {
+			 window.navigator.msSaveOrOpenBlob(file);
+		 } else {
+			 const a = document.createElement('a');
+			 const url = URL.createObjectURL(file);
+			 a.href = url;
+			 a.download = filename;
+			 document.body.appendChild(a);
+			 a.click();
+			 document.body.removeChild(a);
+			 window.URL.revokeObjectURL(url);
+		 }
+	 };
+
+	 const loadFromFile = (file: FileList | null) => {
+		 if (file != null) {
+			 const reader = new FileReader();
+			 reader.readAsText(file.item(0) as Blob);
+			 reader.onload = () => {
+				 jsonMessagesFromString(reader.result);
+			 };
+		 }
+	 };
+
+	 const jsonMessagesFromString = (rawFromFile: string | ArrayBuffer | null) => {
+		 try {
+			 const json = JSON.parse(rawFromFile as string);
+			 store.clearParsedMessages();
+
+			 for (let i = 0; i < json.length; i++) {
+				 json[i].delay = 0;
+				 store.addParsedMessage(json[i]);
+			 }
+		 } catch (Error) {
+			 // eslint-disable-next-line no-alert
+			 alert(Error);
+		 }
+	 };
+
+	 const replaySendMessage = (array: ParsedMessageItem[] | ActMessageItem[], index: number) => {
+		 if (store.isReplay) {
+			 setTimeout(() => {
+				 store.sendMessage(array[index], index);
+				 // eslint-disable-next-line no-param-reassign
+				 index++;
+				 if (index === array.length) {
+					 switchReplay();
+				 } else {
+					 replaySendMessage(array, index);
+				 }
+			 }, array[index].delay);
+		 }
+	 };
+
+	 let switchValue = false;
```

Comment:
if you want some state to persist between renders you should use useState hook
otherwise, everytime component rerenders switchValue will be false
and inisde switchReplay function 	`switchValue = !switchValue` will always be true